cdap-app-fabric/src/main/java/io/cdap/cdap/internal/sysapp/SystemAppEnableExecutor.java

```diff
@@ -136,11 +136,13 @@ private ApplicationWithPrograms deploySystemApp(Arguments arguments) throws Exce
 
   private void startProgram(ProgramId programId) throws Exception {
     try {
-      // TODO(CDAP-16243): Restart program if the application has changed.
-      // do nothing if the program is already running
       ProgramStatus currentStatus = programLifecycleService.getProgramStatus(programId);
-      if (currentStatus != ProgramStatus.STOPPED) {
-        return;
+      // If a system app is already running, it needs to be restarted as app artifact/arguments might have changed.
+      // Restart should trigger running programs with latest version.
+      // TODO(CDAP-16243): Find smarter way to trigger restart of programs rather than always restarting. May be
+      //                   checking if artifact version has changed would be a good start.
+      if (currentStatus == ProgramStatus.RUNNING) {
+        programLifecycleService.stop(programId);
```

Comment:
I believe this can also throw a ConflictException if the program is already stopped. If so, it will be caught, then ignored and the program will never be started.

Should wrap this in a try-catch as well, ignore the conflict, and move on to running it.